javascript/ql/src/semmle/javascript/frameworks/WebSocket.qll

```diff
@@ -69,55 +72,59 @@ module ClientWebSocket {
   }
 
   /**
-   * Gets a methodName that can be used to register a listener for WebSocket messages on a given socket.
+   * A handler that is registered to receive messages from a WebSocket.
    */
-  string getARegisterMethodName(ClientSocket socket) {
-    result = "addEventListener"
-    or
-    socket.isNode() and
-    result = EventEmitter::on()
+  abstract class ReceiveNode extends EventRegistration::Range, DataFlow::FunctionNode {
+    override ClientSocket emitter;
+
+    override string getChannel() { result = channelName() }
   }
 
   /**
-   * A handler that is registered to receive messages from a WebSocket.
-   *
-   * If the registration happens with the "addEventListener" method or the "onmessage" setter property, then the handler receives an event with a "data" property.
-   * Otherwise the handler receives the data directly.
-   *
-   * This confusing API is caused by the "ws" library only mostly using their own API, where event objects are not used.
-   * But the "ws" library additionally supports the WebSocket API from browsers, which exclusively use event objects with a "data" property.
+   * Gets a handler, that is registered using method `methodName` and receives messages send to `emitter`.
    */
-  class ReceiveNode extends EventRegistration::Range, DataFlow::FunctionNode {
-    override ClientSocket emitter;
-    boolean receivesEvent;
+  private DataFlow::FunctionNode getAMessageHandler(ClientWebSocket::ClientSocket emitter, string methodName) {
+    exists(DataFlow::CallNode call |
+      call = emitter.getAMemberCall(methodName) and
+      call.getArgument(0).mayHaveStringValue("message") and
+      result = call.getCallback(1)
+    )
+  }
 
-    ReceiveNode() {
-      exists(DataFlow::CallNode call, string methodName |
-        methodName = getARegisterMethodName(emitter) and
-        call = emitter.getAMemberCall(methodName) and
-        call.getArgument(0).mayHaveStringValue("message") and
-        this = call.getCallback(1) and
-        if methodName = "addEventListener" then receivesEvent = true else receivesEvent = false
-      )
+  /**
+   * A handler that receives a message using the WebSocket API.
+   * The "ws" library implements a superset of the WebSocket API, and this class therefore models both WebSocket and "ws".
```

Comment:
Did you mixup WebSocket/ws here? Dhe docstring is for `ws`, but the class name is for WebSocket... Please double check the characteristic predicates when you resolve this naming/docstring mistake.

Reply:
Yeah, the docstrings could be better.

The class describes the `WebSocket` API, so the class primarily models the `WebSocket` browser library. 

However, the `ws` library implements a superset of the `WebSocket` API, and thus the class is used to model both the `WebSocket` library, and a subset of the `ws` library. (The remainder of the `ws` library is modeled in the `WSReceiveNode` class below).